aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -221,22 +221,8 @@ cudnn_frontend::ExecutionPlan* find(const KeyType& key) {
     return nullptr;
   }
   if (lru_cache_limit) {
-    TORCH_INTERNAL_ASSERT(*(it->second.second) == key, "CUDNN V8 LRU Cache Corrupted (found key mismatches list). Please report a bug to PyTorch.");
-    auto engine_cache_order_size = engine_cache_order.size();
-    auto engine_cache_size = engine_cache.size();
-    TORCH_INTERNAL_ASSERT(engine_cache_order_size == engine_cache_size, "CUDNN V8 LRU Cache Corrupted (found list vs. map size mismatch). Please report a bug to PyTorch.");
     // update most recently accessed
-    auto plan = it->second.first;
-    engine_cache_order.erase(it->second.second);
-    engine_cache_order.push_back(key);
-    engine_cache.erase(key);
-    engine_cache.emplace(key, std::make_pair(plan, --engine_cache_order.end()));
-    // iterator was invalidated by the erase, so we grab it again
-    it = engine_cache.find(key);
-    TORCH_INTERNAL_ASSERT(it->first == *(it->second.second), "CUDNN V8 LRU Cache Corrupted (refresh list vs. map key mismatch). Please report a bug to PyTorch.");
-    TORCH_INTERNAL_ASSERT((long) engine_cache_order.size() <= lru_cache_limit, "CUDNN V8 LRU Cache Corrupted (refresh size exceeds limit: ", lru_cache_limit, " please report a bug to PyTorch.");
-    TORCH_INTERNAL_ASSERT(engine_cache_order.size() == engine_cache_order_size, "CUDNN V8 LRU Cache Corrupted (list size unexpectedly changed). Please report a bug to PyTorch.");
-    TORCH_INTERNAL_ASSERT(engine_cache.size() == engine_cache.size(), "CUDNN V8 LRU Cache Corrupted (cache size unexpectedly changed). Please report a bug to PyTorch.");
+    engine_cache_order.splice(engine_cache_order.begin(), engine_cache_order, it->second.second);
```

Comment:
make engine_cache_order's front most recently, due to `splice(list.begin, list, iter)` better than `splice(prev(list.end()), list, iter)`

Reply:
This is purely to optimize the `engine_cache_order` update, correct? AFAICT it should not change the least-recently-used behavior that was already implemented.

Reply:
Ah, I see, you are changing the implementation to be (front-most-recently-used) from (back-most-recently-used). Understood.